src/helpers/ConfigHelper.ts

```diff
@@ -33,13 +40,186 @@ export interface IntegrationCapabilities {
 const MAX_CATEGORIES = 10;
 
 export class ConfigHelper {
+  /**
+   * convertConfigToVersionOne - converts v0 schema to v1 schema format
+   *
+   * v0 schema example
+   * ---
+   *  promptOptions: {
+   *      acceptButtonText: '',
+   *      cancelButtonText: '',
+   *      actionMessage   : '',
+   *      slidedown: {...}
+   *  }
+   *
+   * v1 schema example
+   * ---
+   * "promptOptions": {
+   *    "slidedown": {
+   *      "enabled": true,
+   *      "autoPrompt": true,
+   *      "acceptButtonText": "",
+   *      "cancelButtonText": "",
+   *      "actionMessage": "",
+   *      "...",
+   *    }
+   * }
+   * @param  {any} promptOptions
+   * @returns AppUserConfigPromptOptions
+   */
+  private static convertConfigToVersionOne(promptOptions: any) : AppUserConfigPromptOptions {
+    if (!this.isPromptOptionsVersion0(promptOptions)) {
+      throw new Error("Cannot convert promptOptions config to version 1 since config is not version 0");
+    }
+
+    if (!promptOptions.slidedown) {
+      promptOptions.slidedown = {};
+    }
+
+    promptOptions.slidedown.acceptButtonText = promptOptions.acceptButton  || promptOptions.acceptButtonText;
+    promptOptions.slidedown.cancelButtonText = promptOptions.cancelButton  || promptOptions.cancelButtonText;
+    promptOptions.slidedown.actionMessage    = promptOptions.actionMessage;
+
+    return promptOptions;
+  }
+
+  private static convertConfigToVersionTwo(slidedownConfig: any) : SlidedownOptions {
+    if (!this.isSlidedownConfigVersion1(slidedownConfig)) {
+      throw new Error("Cannot convert slidedown config to version 2 since config is not version 1");
+    }
+
+    // determine if the slidedown is category type or regular push
+    const promptType = PromptsHelper.isCategorySlidedownConfiguredVersion1(slidedownConfig) ?
+      DelayedPromptType.Category : DelayedPromptType.Push;
+
+    let positiveUpdateButton, negativeUpdateButton: string|undefined;
+    if (promptType === DelayedPromptType.Category) {
+      positiveUpdateButton = slidedownConfig.categories?.positiveUpdateButton;
+      negativeUpdateButton = slidedownConfig.categories?.negativeUpdateButton;
+    }
+
+    return {
```

Comment:
This would clobber the `prompts` array the dev had both defined. Example:
```js
slidedown: {
}
prompts: [
]
```

We should append their existing `prompts` if they have one.